docs/forms-in-wagtail.md

```diff
@@ -0,0 +1,44 @@
+# Including forms into Wagtail page context
```

Comment:
I like this documentation although I might leave out the first two paragraphs which are, although perhaps fairly critical of Wagtail's form support, more internal details of our development process and less appropriate for developers trying to use the code.